### ISSUE TITLE
mobile AMO allows installing incompatible extensions

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-frontend#2087]
Steps to reproduce:
1. On Desktop, select an Android add-on that is not compatible with your platform i.e. https://addons.mozilla.org/en-US/android/addon/cloud-viewer
2. Click "View Mobile Site"
3. Install the add-on
4. Go in about:addons Extensions tab and see your add-on

Expected results:
Add-on is installed and disabled. Reason of disabling the add-on is that it is incompatible with your platform.

Actual results:
Reason of disabling the add-on is that it is incompatible with your version of Firefox.

Notes/Issues:
 - Installing the add-on on device, using the same version of Firefox, as before, will work without any messages.

Verified on FF52(Win 7, Android 6.0.1) on AMO-prod.
Video for this issue:
![wrongmsg](https://cloud.githubusercontent.com/assets/15685960/24036706/728ff016-0b03-11e7-9410-b7987b6f4e1e.gif)

**Comment by muffinresearch** [comment]
@ValentinaPC note that the discovery pane is *not* the list of addons in about:addons. It is merely the page with the curated list of addons which looks like this:

![add-ons_manager](https://cloud.githubusercontent.com/assets/1514/24037191/d9ec2f86-0af4-11e7-8827-782dc83c61d0.png)

**Comment by ValentinaPC** [comment]
> Actually fixing the incorrect message in about:addons#Extensions would require a bugzilla bug to be raised.

Logged a Bugzilla issue under: https://bugzilla.mozilla.org/show_bug.cgi?id=1349135

**Comment by designakt** [comment]
To prevent this problem from arising, we should implement @muffinresearch suggestion.
> I think the bug we can action here is that the mobile page should have a message about platform compatibility and it should follow desktop in not allowing installation for an incorrect platform.

**Comment by designakt** [comment]
@pwalm can you please take this - you new desktop might already have a UI for this.

**Comment by pwalm** [comment]
- No install button
- Replace install button with "Not available for your platform" copy from desktop site (will have to break it into two lines)
- Align copy to the right

**Comment by muffinresearch** [comment]
@tofumatt would you have a chance to look at this? It would be good to add this messaging to avoid confusion especially if the platforms allow installation of incorrect add-ons.

@jvillalobos mentioned developers are also getting confused by this when viewing /firefox/ from android.

**Comment by tofumatt** [comment]
> developers are also getting confused by this when viewing /firefox/ from android.

Solution to that might be to display a little dismissible warning message at the top of the page saying something like "You're viewing add-ons for Firefox Desktop; tap here to view Android add-ons." or whatever. I'll file a separate issue.

I think we'll still need that after web extensions so it's worth doing. I don't think an automatic redirect is good UX but a prominent message is 👍

**Comment by muffinresearch** [comment]
@tofumatt any updates on this one? Are we still waiting on UX?

**Comment by tofumatt** [comment]
I don't think so but I'll check. If not I should be able to take it. Sorry 😓

**Comment by kumar303** [comment]
@tofumatt I think this got fixed in https://github.com/mozilla/addons-frontend/pull/2934, right?

**Comment by tofumatt** [comment]
Yes, it should be!

**Comment by ValentinaPC** [comment]
Verified as fixed on AMO-dev FF55(Win 7).
Incompatible add-ons with you FF version are disabled in mobile view / new desktop pages and a red banner is displayed.
Postfix screenshot:
![inst](https://user-images.githubusercontent.com/15685960/29617025-074f67e6-881c-11e7-8b98-0ed1aa02b580.png)
